asreviewcontrib/datatools/dedup.py

```diff
@@ -76,7 +77,7 @@ def drop_duplicates_by_similarity(
         stopwords_regex = re.compile(rf'\b{"\\b|\\b".join(stopwords_set)}\b')
         s = s.str.replace(stopwords_regex, '', regex=True)
 
-    duplicated = (s.duplicated()) & (s.notnull())
+    duplicated = [False] * len(s)
```

Comment:
I prefer `[False for _ in range(len(s))]`. In this case it does not really matter, but if you put a dictionary instead of `False` inside the list, then your list will actually contain references to _the same_ dictionary instead of copies of dictionaries.

Reply:
Yes, but in this case the content is just `False`, and in the line 101 this value will be updated to `True`, there will be never a dictionary here. And in this case, the current solution is faster. See:

```python
In [1]: %timeit [False] * 10_000_000
29.6 ms ± 2.81 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [2]: %timeit [False for _ in range(10_000_000)]
187 ms ± 2.28 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```